Buildings/Applications/DHC/EnergyTransferStations/CoolingIndirect.mo

```diff
@@ -0,0 +1,287 @@
+    "Gain of controller"
+    annotation(Dialog(tab="Controller"));
+  parameter Modelica.SIunits.Time Ti(min=Modelica.Constants.small)=0.5
+    "Time constant of integrator block"
+     annotation (Dialog(tab="Controller"));
+  parameter Modelica.SIunits.Time Td(min=0)=0.1
+    "Time constant of derivative block"
+     annotation (Dialog(tab="Controller"));
+  parameter Real yMax(start=1)=1
+   "Upper limit of output"
+    annotation(Dialog(tab="Controller"));
+  parameter Real yMin=0
+   "Lower limit of output"
+    annotation(Dialog(tab="Controller"));
+  parameter Real wp(min=0) = 1
+   "Set-point weight for Proportional block (0..1)"
+    annotation(Dialog(tab="Controller"));
+  parameter Real wd(min=0) = 0
+   "Set-point weight for Derivative block (0..1)"
+    annotation(Dialog(tab="Controller"));
+  parameter Real Ni(min=100*Modelica.Constants.eps) = 0.9
+    "Ni*Ti is time constant of anti-windup compensation"
+    annotation(Dialog(tab="Controller"));
+  parameter Real Nd(min=100*Modelica.Constants.eps) = 10
+    "The higher Nd, the more ideal the derivative block"
+    annotation(Dialog(tab="Controller"));
+  parameter Modelica.Blocks.Types.InitPID initType=
+    Modelica.Blocks.Types.InitPID.DoNotUse_InitialIntegratorState
+    "Type of initialization (1: no init, 2: steady state, 3: initial state, 4: initial output)"
+    annotation(Dialog(group="Initialization",tab="Controller"));
+  parameter Real xi_start=0
+    "Initial or guess value value for integrator output (= integrator state)"
+    annotation (Dialog(group="Initialization",tab="Controller"));
+  parameter Real xd_start=0
+    "Initial or guess value for state of derivative block"
+    annotation (Dialog(group="Initialization",tab="Controller"));
+  parameter Real yCon_start=0
+    "Initial value of output from the controller"
+    annotation(Dialog(group="Initialization",tab="Controller"));
+  parameter Boolean reverseAction = true
+    "Set to true for throttling the water flow rate through a cooling coil controller"
+    annotation(Dialog(tab="Controller"));
+
+  Buildings.Fluid.HeatExchangers.PlateHeatExchangerEffectivenessNTU hex(
+    redeclare package Medium1 = Medium,
+    redeclare package Medium2 = Medium,
+    m1_flow_nominal=m1_flow_nominal,
+    m2_flow_nominal=m2_flow_nominal,
+    dp1_nominal=dp1_nominal,
+    dp2_nominal=dp2_nominal,
+    configuration=Buildings.Fluid.Types.HeatExchangerConfiguration.CounterFlow,
+    use_Q_flow_nominal=true,
+    Q_flow_nominal=Q_flow_nominal,
+    T_a1_nominal=T_a1_nominal,
+    T_a2_nominal=T_a2_nominal) "Indirect cooling heat exchanger"
+    annotation (Placement(transformation(
+        extent={{10,10},{-10,-10}},
+        rotation=180,
+        origin={30,0})));
+
+  Modelica.Blocks.Interfaces.RealInput TSet "Setpoint temperature"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
+
+  Buildings.Controls.Continuous.LimPID con(
+    controllerType=Modelica.Blocks.Types.SimpleController.PID,
```

Comment:
`final controllerType=controllerType,`

Also, since all these parameters of `con` have been propagated, they can be finalized here.

Reply:
Final added to all parameters for next commit.